odoo/addons/base/models/res_company.py

```diff
@@ -221,7 +220,14 @@ def _compute_uninstalled_l10n_module_ids(self):
             company.uninstalled_l10n_module_ids = self.env['ir.module.module'].browse(mapping.get(company.country_id.id))
 
     def install_l10n_modules(self):
-        return self.uninstalled_l10n_module_ids.button_immediate_install()
+        companies = self.filtered('uninstalled_l10n_module_ids')
+        is_ready_and_not_test = (
+            not tools.config['test_enable']
+            and (self.env.registry.ready or not self.env.registry._init)
+        )
+        if companies and is_ready_and_not_test:
+            return companies.uninstalled_l10n_module_ids.button_immediate_install()
+        return is_ready_and_not_test
```

Comment:
Let me answer your `why?` before you ask it. Suppose a new company is created with country and localisation gets installed and COA is loaded and then I try to create another company with same country, l10n modules are already installed. So in the `account` module where we inherit the method just misses out the COA because it would not get any return `value`.